### ISSUE TITLE
khal won't build on master because of failing tests

### ISSUE THREAD

**Issue by DamienCassou** [NixOS/nixpkgs#205014]
### Steps To Reproduce
1. switch to the `master` branch
2. `nix-build -A khal`

### Build log
```
>       assert create_timezone(bogota, atime, atime).to_ical().split(b'\r\n') == vbogota
>       assert create_timezone(bogota, atime, atime).to_ical().split(b'\r\n') == vbogota
E       AssertionError: assert [b'BEGIN:VTIM...M:-0400', ...] == [b'BEGIN:VTIM...AME:-05', ...]
E       AssertionError: assert [b'BEGIN:VTIM...M:-0400', ...] == [b'BEGIN:VTIM...AME:-05', ...]
E         At index 4 diff: b'TZNAME:-05' != b'RDATE:20380118T221407'
E         At index 4 diff: b'TZNAME:-05' != b'RDATE:20380118T221407'
E         Right contains one more item: b''
E         Right contains one more item: b''
E         Use -v to get more diff
E         Use -v to get more diff


tests/vtimezone_test.py:81: AssertionError
tests/vtimezone_test.py:81: AssertionError
=========================== short test summary info ============================
=========================== short test summary info ============================
FAILED tests/event_test.py::test_event_no_dst - AssertionError: assert ('VCAL...
FAILED tests/event_test.py::test_event_no_dst - AssertionError: assert ('VCAL...
FAILED tests/vtimezone_test.py::test_bogota - AssertionError: assert [b'BEGIN...
FAILED tests/vtimezone_test.py::test_bogota - AssertionError: assert [b'BEGIN...
====== 2 failed, 302 passed, 3 deselected, 1 xfailed, 2 xpassed in 3.05s =======
====== 2 failed, 302 passed, 3 deselected, 1 xfailed, 2 xpassed in 3.05s =======
```

### Notify maintainers

@gebner @fabaff 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.10-200.fc36.x86_64, Fedora Linux, 36 (Workstation Edition), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `""`
 - nixpkgs: `/home/cassou/Documents/projects/nix-system/nixpkgs`
```

**Comment by toastal** [comment]
Now it doesn't build on unstable as well. This package is pretty fussy with recurring issues.

**Comment by lucc** [comment]
This is probably the upstream issue: https://github.com/pimutils/khal/issues/1205

**Comment by mdarocha** [comment]
Note that this error was fixed by https://github.com/NixOS/nixpkgs/pull/207992, this issue can probably be closed.

**Comment by lucc** [comment]
`nix build nixpkgs#khal` indeed succeeds. Upstream also has an open PR for this issue: https://github.com/pimutils/khal/pull/1216

Should  we keep this issue open to undo #207992 when the upstream PR is merged or just close this issue here?

**Comment by DamienCassou** [comment]
I think closing is fine.

**Comment by mdarocha** [comment]
It seems like it started to fail again, with a different set of tests:

```sh
error: builder for '/nix/store/bbbcrp1jxfmm1sw2nz6h0n03mjwnhwjm-khal-0.10.5.drv' failed with exit code 1;
       last 10 log lines:
       > FAILED tests/parse_datetime_test.py::test_leap_year - AssertionError: assert b'BEGIN:VEVEN...ND:VEVENT\r\n' == b'BEGIN:VEVEN...ND...
       > FAILED tests/parse_datetime_test.py::test_description - AssertionError: assert b'BEGIN:VEVEN...ND:VEVENT\r\n' == b'BEGIN:VEVEN...ND...
       > FAILED tests/parse_datetime_test.py::test_repeat_floating - AssertionError: assert ('VEVENT', b'..., frozenset()) == ('VEVENT', b'..., ...
       > FAILED tests/parse_datetime_test.py::test_repeat_localized - AssertionError: assert ('VEVENT', b'..., frozenset()) == ('VEVENT', b'..., ...
       > FAILED tests/parse_datetime_test.py::test_alarm - AssertionError: assert b'BEGIN:VEVEN...ND:VEVENT\r\n' == b'BEGIN:VEVEN...ND...
       > FAILED tests/parse_datetime_test.py::test_description_and_location_and_categories - AssertionError: assert b'BEGIN:VEVEN...ND:VEVENT\r\n' == b'BEGIN:VEVEN...ND...
       > FAILED tests/vtimezone_test.py::test_berlin - AssertionError: assert b'BEGIN:STANDARD\r\nDTSTART;VALUE=DATE-TIME:20141026...
       > FAILED tests/vtimezone_test.py::test_berlin_rdate - AssertionError: assert b'BEGIN:STANDARD\r\nDTSTART;VALUE=DATE-TIME:20141026...
       > ===== 19 failed, 282 passed, 6 deselected, 1 xfailed, 2 xpassed in 55.76s ======
       > /nix/store/chilfhdcsnmwjl7igrw26j1lrc0zar35-stdenv-linux/setup: line 1570: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/bbbcrp1jxfmm1sw2nz6h0n03mjwnhwjm-khal-0.10.5.drv'.
```

Maybe we should just disable testing on this package?

**Comment by toastal** [comment]
Now I have issue with `urwid` not building either as a dependency.

@mdarocha I've had tests disabled for a while to deal with the test failures and it's ran alright.

**Comment by toastal** [comment]
This is extra annoying because you can't override it with an overlay.

```
final: prev: {
   khal = prev.khal.overrideDerivation (old: { doCheck = false; });
}
```

because the Python packages use `rec`.

**Comment by mdarocha** [comment]
You can, using `overridePythonAttrs`

```nix
  khal = super.khal.overridePythonAttrs (_: {
    doCheck = false;
  });
  todoman = super.todoman.overridePythonAttrs (_: {
    doCheck = false;
  });
```

note: I had to add the same overlay for `todoman`, since it also has problems with tests.

Both problems seem to be caused by an update to `icalendar` package

**Comment by fabaff** [comment]
> Now I have issue with urwid not building either as a dependency.

`urwid` was [fixed](https://github.com/NixOS/nixpkgs/pull/211812) recently.

**Comment by lucc** [comment]
Should be fixed by #227861
